src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -456,7 +459,30 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     requestFilterPromise = requestFilterPromise || Promise.resolve(pkRequest);
     requestFilterPromise
       .then(updatedRequest => {
-        this._videoElement.src = updatedRequest.url;
+        if (this._config.useSourceTag) {
+          const source = document.createElement('source');
+          const mimetype = this._sourceObj.mimetype.toLowerCase();
+          source.setAttribute('src', updatedRequest.url);
+          source.setAttribute('type', mimetype);
+
+          if (this._config.useMediaOptionAttribute) {
+            NativeAdapter._logger.debug('this._config.abrEwmaDefaultEstimate', this._config.abrEwmaDefaultEstimate);
+            const options = {};
+            options.option = {};
+            if (this._config.abrEwmaDefaultEstimate) {
+              options.option.adaptiveStreaming = {};
+              options.option.adaptiveStreaming.bps = {
+                start: this._config.abrEwmaDefaultEstimate
+              };
+            }
+            NativeAdapter._logger.debug('options', options);
+            const mediaOption = encodeURI(JSON.stringify(options));
```

Comment:
what needs to be encoded here ? is it only the src ? then you can run encode on the src instead of on the entire object

Reply:
according to LG entire object